src/util/live_performance.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+
+import window from './window.js';
+import {version as sdkVersion} from '../../package.json';
+import type Projection from '../geo/projection/projection.js';
+import {
+    isMapboxHTTPStyleURL,
+    isMapboxHTTPTileJSONURL,
+    isMapboxHTTPSpriteURL,
+    isMapboxHTTPFontsURL
+} from './mapbox.js';
+
+type LivePerformanceMetrics = {
+    counters: Array<Object>,
+    metadata: Array<Object>,
+    attributes: Array<Object>
+};
+
+export type LivePerformanceData = {
+    interactionRange: [number, number],
+    width: number,
+    height: number,
+    terrainEnabled: boolean,
+    fogEnabled: boolean,
+    projection: Projection,
+    renderer: ?string,
+    vendor: ?string
+};
+
+function categorize(arr, fn) {
+    const obj = {};
+    if (arr) {
+        for (const item of arr) {
+            const category = fn(item);
+            if (obj[category] === undefined) {
+                obj[category] = [];
+            }
+            obj[category].push(item);
+        }
+    }
+    return obj;
+}
+
+function getTransferRangePerResourceType(resourceTimers) {
+    const obj = {};
+    if (resourceTimers) {
+        for (const category in resourceTimers) {
+            if (category !== 'other') {
+                for (const timer of resourceTimers[category]) {
+                    const min = `${category}TransferStart`;
```

Comment:
Since it's the time range taken between:
- [startTime](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/startTime)
>"resource" - returns the [timestamp](https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp) immediately before the browser [starts fetching the resource](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/fetchStart).
- [responseEnd](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/responseEnd):
>The responseEnd read-only property returns a [timestamp](https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp) immediately after the browser receives the last byte of the resource or immediately before the transport connection is closed, whichever comes first

maybe a better name would be `ResolveStart` `ResolveEnd`, and add `Range` to denote that it's a range of values for each of these categories?

In the network tab, it's the time named "network transfer":
<img width="532" alt="Screen Shot 2022-11-04 at 12 47 21 PM" src="https://user-images.githubusercontent.com/7061573/200062292-92b11e78-3268-4fd5-a708-e117e40cb258.png">

Reply:
Renamed in https://github.com/mapbox/mapbox-gl-js/pull/12343/commits/9b80532eeb00ccefbb4b7b7d9b9ca496cf40aab4.